--- a/soh/soh/z_message_OTR.cpp
+++ b/soh/soh/z_message_OTR.cpp
@@ -15,6 +15,7 @@ extern "C" MessageTableEntry* sGerMessageEntryTablePtr;
 extern "C" MessageTableEntry* sFraMessageEntryTablePtr;
 extern "C" MessageTableEntry* sJpnMessageEntryTablePtr;
+extern "C" MessageTableEntry* sSpaMessageEntryTablePtr;
 extern "C" MessageTableEntry* sStaffMessageEntryTablePtr;
 
@@ -88,6 +89,9 @@ extern "C" void OTRMessage_Init() {
     if (sJpnMessageEntryTablePtr == NULL) {
         sJpnMessageEntryTablePtr = OTRMessage_LoadTable("text/jpn_message_data_static/jpn_message_data_static", false);
     }
+    if (sSpaMessageEntryTablePtr == NULL) {
+        sSpaMessageEntryTablePtr = OTRMessage_LoadTable("text/spa_message_data_static/spa_message_data_static", false);
+    }
 
     // Note: Make sure this loads after PAL nes_message_data_static, so that message 0xFFFC is definitely loaded if it
